client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/index.tsx

```diff
@@ -149,10 +152,11 @@ const designSetup: Step = function DesignSetup( { navigation, flow } ) {
 
 	const getEventPropsByDesign = ( design: Design ) => ( {
 		theme: design.recipe?.stylesheet,
-		template: design.template,
+		template: design?.template,
 		is_premium: design?.is_premium,
-		flow,
+		flow: showGeneratedDesigns ? 'generated-design-step' : flow,
 		intent: intent,
+		...( design?.recipe?.patternIds && { pattern_ids: design.recipe.patternIds.join( ',' ) } ),
```

Comment:
I'm not sure but I prefer to track the id or title of the recipe and it might be easier to know which recipe is more popular. If we use pattern ids, I'm not sure if it's good for analysis or not. But maybe the patterns are more important than the recipe. What do you think?

Reply:
I think it's good to keep `pattern_ids`, but I've also added `slug` as you suggested 🙂